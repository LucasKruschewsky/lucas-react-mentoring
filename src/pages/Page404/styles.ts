import styled from 'styled-components';

export const Container404 = styled.div`
  padding-top: var(--offset-header);
  min-height: 75.5vh;

  .button-link {
    text-decoration: none;
  }

  div {
    background-color: rgb(var(--white));
    padding: 2rem;
    border-radius: 4px;
    text-align: center;

    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;

    h1 {
      font-size: var(--title-1);
      margin-bottom: 1rem;
    }

    .subtitle {
      font-size: var(--body-1);
    }

    p {
      margin-bottom: 1rem;
    }

    button {
      width: fit-content;
    }
  }
`;
