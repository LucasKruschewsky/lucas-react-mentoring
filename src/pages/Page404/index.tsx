import AppButton from 'Global/styled/AppButton';
import AppContainer from 'Global/styled/AppContainer';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { Container404 } from './styles';

const Page404: React.FunctionComponent<{}> = () => (
  <AppContainer>
    <Container404>
      <div>
        <h1>404</h1>
        <p className="subtitle">Sorry, page could not be found.</p>
        <p>Either something went wrong, or the page does not exist anymore.</p>
        <Link className="button-link" to="/">
          <AppButton>Back to Homepage</AppButton>
        </Link>
      </div>
    </Container404>
  </AppContainer>
);

export default Page404;
