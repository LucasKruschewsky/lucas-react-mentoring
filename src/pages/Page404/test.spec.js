import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Page404 from '.';

describe('Page404', () => {
  it('Snapshot test', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Page404 />
      </MemoryRouter>
    ).toJSON;

    expect(tree).toMatchSnapshot();
  });
});
