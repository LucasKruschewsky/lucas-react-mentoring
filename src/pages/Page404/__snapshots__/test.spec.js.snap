// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page404 Snapshot test 1`] = `[Function]`;
